Paramedic Void Suit update

Velken opened the pull request:
<!-- Guidelines: https://docs.spacestation14.io/en/getting-started/pr-guideline -->

## About the PR
<!-- What did you change? -->
Moved the paramedic void suit outof the softsuit yml and moved it into the hardsuit yml
Made it restricted to medical (to be in line with the other deparment-specific hardsuits)
Added flashlight to its helmet
Added rotating blue light with siren sound to the suit (similar to the justice helmet)

## Why / Balance
<!-- Discuss how this would affect game balance or explain why it was changed. Link any relevant discussions or issues. -->
It was the only hardsuit (to my knowledge) that didn't have flashlight.
And being such a limited item per station, I though it would be nice to give it an ambulance siren.

## Technical details
<!-- Summary of code changes for easier review. -->
Moved ClothingOuterHardsuitVoidParamed and ClothingHeadHelmetVoidParamed from softsuit/EVA to the hardsuit .yml's
Added ambulancesiren.ogg and updated attributions .yml in its folder


## Media
<!-- Attach media if the PR makes ingame changes (clothing, items, features, etc). 
Small fixes/refactors are exempt. Media may be used in SS14 progress reports with credit. -->
https://youtu.be/XnXVAvqczww

Note: the graphics in the video are outdated, the light is now from the side of the helmet instead of the antenna

## Requirements
<!-- Confirm the following by placing an X in the brackets [X]: -->
- [X] I have read and am following the [Pull Request and Changelog Guidelines](https://docs.spacestation14.com/en/general-development/codebase-info/pull-request-guidelines.html).
- [X] I have added media to this PR or it does not require an ingame showcase.
<!-- You should understand that not following the above may get your PR closed at maintainer’s discretion -->

## Breaking changes
<!-- List any breaking changes, including namespaces, public class/method/field changes, prototype renames; and provide instructions for fixing them.
This will be posted in #codebase-changes. -->
Prototypes parenting off ClothingHeadHardsuitWithLightBase need to be reparented to use both ClothingHeadHardsuitBase AND ClothingHeadSuitWithLightBase

**Changelog**
<!-- Add a Changelog entry to make players aware of new features or changes that could affect gameplay.
Make sure to read the guidelines and take this Changelog template out of the comment block in order for it to show up.
Changelog must have a :cl: symbol, so the bot recognizes the changes and adds them to the game's changelog. -->
<!--
:cl:
- add: Added fun!
- remove: Removed fun!
- tweak: Changed fun!
- fix: Fixed fun!
-->

:cl:
- add: Paramedic Void Suit now has flashlight and an ambulance siren.
- tweak: Paramedic Void Suit is now restricted to Medical.

slarticodefast commented:
Isn't the difference between a soft and hardsuit the damage protection and slowdown?
So it would still be a softsuit since it gives no protection besides heat and cold.

Velken commented:
> Isn't the difference between a soft and hardsuit the damage protection and slowdown? So it would still be a softsuit since it gives no protection against brute damage.

The old code has a comment that says
"Despite having resistances and looking like a hardsuit, this parents off the EVA suit so it goes here."
The old one already has slowdown and protection values, just not for brute damage.
Plus, it looks and behaves more like a hardsuit than a softsuit.

UbaserB commented:
The light shouldn’t be on the antennae. The antennae is way too thin for that and serves a different function.

Velken commented:
> The light shouldn’t be on the antennae. The antennae is way too thin for that and serves a different function.
> 
> Also, should probably have a custom icon for the siren action.

Where would you suggest the light be then?

UbaserB commented:
![IMG_6898](https://github.com/user-attachments/assets/3bf0f6f4-7ff8-4bdc-90bb-e7b380e1e297)

Velken commented:
> ![IMG_6898](https://private-user-images.githubusercontent.com/134914314/406733739-3bf0f6f4-7ff8-4bdc-90bb-e7b380e1e297.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3Mzc4OTc1OTMsIm5iZiI6MTczNzg5NzI5MywicGF0aCI6Ii8xMzQ5MTQzMTQvNDA2NzMzNzM5LTNiZjBmNmY0LTdmZjgtNGJkYy05MGJiLWU3YjM4MGUxZTI5Ny5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjUwMTI2JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI1MDEyNlQxMzE0NTNaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT1iYTQ5MzRhMjliMDhjZmU4YzU3YTU4N2Y5ZTIxNGEwZTU3M2VmMzIxM2UwNjRiMTA1ZGY2OTJkNjVlOTBjNTYxJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.Z2Eh0J69zJejW7ptRSS-bpB1ljclUFjvmPNWRJQO1sk)

ooooh, i like that, i'll update the sprite after in a bit and also make a custom icon for the siren

Velken commented:
changed the light placement
but i couldn't figure out how to change the action bar icon for the siren

Velken commented:
fixed the siren icon, thanks to, Milon, KeronSHB and Orks (Beck T) on discord for helping out

Velken commented:
made siren icon BigAction size
![image](https://github.com/user-attachments/assets/b1676d23-9f28-4dbb-95eb-934a1858cde6)

UbaserB commented:
> made siren icon BigAction size

No mixels. Fix the sprite size to be consistent with the other actions (32x32)

walksanatora commented:
so I listened to the alarm siren and it is... even more annoying then the justice helm. idk how it sounds in-game but it is LOUD in your video. but I think it needs a diffrent sound.

Velken commented:
> > made siren icon BigAction size
> 
> No mixels. Fix the sprite size to be consistent with the other actions (32x32)

the icon-siren is 32x32, but if you don't like it, i can try make a new one.


> so I listened to the alarm siren and it is... even more annoying then the justice helm. idk how it sounds in-game but it is LOUD in your video. but I think it needs a diffrent sound.

On my end they are the same volume as the justice helmet. If you have a suggestion for a different sound that has the proper licenses we can use, feel free to link it.

UbaserB commented:
icon-siren should look like it is 32x32 because you're bunching the pixels together to give the illusion of it being lower res

Velken commented:
> icon-siren should look like it is 32x32 because you're bunching the pixels together to give the illusion of it being lower res
would this be better?

![image](https://github.com/user-attachments/assets/60aab903-03c8-4a0a-a8c7-39ab5245997d)

![image](https://github.com/user-attachments/assets/574c9f96-523c-46ce-809f-da501c3301c4)

UbaserB commented:
the sprites a bit weird but yes

Velken commented:
> the sprites a bit weird but yes
in-game look

![image](https://github.com/user-attachments/assets/9ad7f1f5-a17d-4592-bb98-8b8de43d9aa9)

superjj18 commented:
From my limited understanding a void suit, compared to a hardsuit, allows faster movement is smaller and thus easier to transport in a bag.

Velken commented:
As I mentioned in a previous comment, from my understanding, it was only in the softsuits YML because it parented off the EVA suit,

The RD's suit is on the hardsuits YML

Now that you mentioned being able to put it in a bag, noticed I accidently removed the ability to put it in one, so i'll do a commit fixing that.

I am not alreatering any of the names, armor or speed modifiers that the suit has, am simply moving it to the hardsuiits YML because my changes to have it have a flashlight changes the parent to hardsuit instead of eva suit.

ScarKy0 reviewed (changes requested):
Just some small things

ScarKy0 commented:
Will still need an official arttainer review

ScarKy0 commented:
also CL was lacking a space, unsure if that matters but i fixed it for you

Velken commented on Resources/Prototypes/Entities/Clothing/Head/hardsuit-helmets.yml:
```diff
@@ -342,6 +342,46 @@
+        Radiation: 0.75
+  - type: PointLight
+    radius: 7
+    energy: 3
```
reduced the light as suggested
changed it to cyan (00ffff) and the helmet light sprite too

![image](https://github.com/user-attachments/assets/57ae11f0-9f40-4c56-9af0-5334de504150)

![image](https://github.com/user-attachments/assets/578112ec-f924-4a0b-926d-8d520c603bd2)

![image](https://github.com/user-attachments/assets/2162aca6-3d81-4409-b88b-a855f3d7ec22)

![image](https://github.com/user-attachments/assets/61825337-6afc-4c99-a7e7-a8d3c78804c5)

ScarKy0 reviewed (approved):
Get approved nerd. Will need an arttainer approval still however.

slarticodefast commented on Resources/Prototypes/Entities/Clothing/Head/hardsuit-helmets.yml:
```diff
@@ -342,6 +342,47 @@
 
+#Paramedic Void Helmet
+- type: entity
+  parent: ClothingHeadHardsuitWithLightBase
```
by reparenting the paramedic suit to the hardsuit you are changing its pressure protection stats
instead turning it into a hardsuit I would split off all the components responsible for the helmet light into a new base parent and have both the hardsuits and softsuits inherit that when needed

slarticodefast commented:
I also think the siren sound is a little hard on the ears. Could you maybe try finding another sound effect for it?

Velken commented:
> I also think the siren sound is a little hard on the ears. Could you maybe try finding another sound effect for it?

hopefully the new sound is better

Aeshus commented on Resources/Textures/Clothing/OuterClothing/Hardsuits/paramed.rsi/icon-siren.png:
**question:** Where are these sprites from? I wasn't easily able to find this sprite from paradise, so I'm assuming this may be from a different source?

If they are from another source, you'll need to add it to the RSI's meta.json's copyright field.

Velken replied:
i just edited the existing sprite for the paramedic void suit myself, same for the siren icon

Aeshus commented on Resources/Textures/Clothing/OuterClothing/Hardsuits/paramed.rsi/meta.json:
```diff
@@ -10,6 +10,20 @@
+    {
+      "name": "icon-siren"
+    },
+	{
+      "name": "icon-siren-light",
+	  "delays": [
+        [
+          0.1,
+          0.1,
+          0.1,
+          0.1
+        ]
+	  ]
```
**nitpick:** There seems to be weird indentation here, so please make sure it's fixed. 

Also, some of the lines have tabs which we don't use -- please use spaces instead (just use the number of spaces to match the indentation of the rest of the file).

Velken replied:
that indentention is copied from justice helmet, since i couldn't get the animated icon to work, i will just remove that sprite state

Aeshus commented on Resources/Prototypes/Entities/Clothing/OuterClothing/hardsuits.yml:
```diff
@@ -471,6 +471,85 @@
+  - type: Sprite
+    sprite: Clothing/OuterClothing/Hardsuits/paramed.rsi
+    layers:
+      - state: icon
```
**nitpick:** This line is indented incorrectly.

Aeshus reviewed (changes requested):
Thanks for the PR!

I just had some nitpicks (so original of me). I also had a question for where your new sprites are from, as you must attribute them all.

Enjoy!

Velken commented:
created ClothingHeadEVAHelmetWithLightBase in base_clothinghead.yml
replaced ambulancesiren.ogg (again, this time with better fading on the audio loop)
credited myself for the siren icon art
reverted parents to original (but kept it in hardsuits yml, since void suits are light hardsuits) - [ the adoption was revoked 😢 ]
fixed indentations
removed un-used sprite and sprite state

Velken commented:
this is what the current ambulance siren sounds like, I will keep looking for a better one, but is not easy 😢 
(video recorded with both game and computer sound at 100%)

https://github.com/user-attachments/assets/8fd29cdb-fed5-4158-a6ce-bcca3be4bfbe

Velken commented:
final final (for real this time???) siren sound effect

https://github.com/user-attachments/assets/4a6ce673-3f46-455d-9989-0c750c7b78a8

Velken commented:
latest siren:

https://github.com/user-attachments/assets/b8a42ff6-7eeb-49b2-8afd-ab1b37013d90

Aeshus commented on Resources/Prototypes/Entities/Clothing/Head/base_clothinghead.yml:
```diff
@@ -230,6 +230,58 @@
+    radiatingBehaviourId: radiating
+  - type: LightBehaviour
+    behaviours:
+      - !type:FadeBehaviour
+        id: radiating
+        interpolate: Linear
+        maxDuration: 2.0
+        startValue: 3.0
+        endValue: 2.0
+        isLooped: true
+        reverseWhenFinished: true
+      - !type:PulseBehaviour
+        id: blinking
+        interpolate: Nearest
+        maxDuration: 1.0
+        minValue: 0.1
+        maxValue: 2.0
+        isLooped: true
```
**nitpick:** These are indented incorrectly. We align the dash with the start of the line above it for some reason (I love editorconfig...).

Aeshus commented on Resources/Prototypes/Entities/Clothing/OuterClothing/hardsuits.yml:
```diff
@@ -471,6 +471,84 @@
+    containers:
+      toggleable-clothing: !type:ContainerSlot {}
+
+- type: entity
+  id: ActionToggleParamedicSiren
+  name: Toggle Paramedic Siren
+  description: Toggles the paramedic siren on and off.
+  components:
+  - type: InstantAction 
+    icon:
+      sprite: Clothing/OuterClothing/Hardsuits/paramed.rsi
+      state: icon-siren
+    useDelay: 1
+    itemIconStyle: BigAction
+    event: !type:ToggleActionEvent
```
**issue (non-blocking):** I'm not sure if the action should be in this file.

I notice that the Justice Helm co-locates the action and the helmet, though a lot of other miscellaneous actions are all put within Actions/types.yml.

Maybe another maintainer can give input on where to store the action, though I'm leaning more towards putting it with all the other actions.

Velken replied:
I actually agree, that it should probably go with Actions/types, will move to there

Aeshus commented on Resources/Textures/Clothing/Head/Hardsuits/paramedhelm.rsi/icon-flash.png:
**question:** Are these light variant sprites from paradise station or do they come from another source?

Velken replied:
sorry if I wasn't clear before: the light variations i made myself (using the normal helmet icon and just adding a light cyan brush on top)

Aeshus replied:
Always make sure to modify the copyright field of the appropriate RSI's meta.json when adding/modifying any sprites.

Aeshus reviewed (changes requested):
Thanks for making the changes I requested.

I just had a few more nitpicks and questions about where the light variant sprites are from.

Velken commented:
attributions updated (turns out, i have to give myself credit when i make stuff)
fixed identation
moved action to Action/types.yml

Aeshus reviewed (commented):
Thanks for the PR!

Looks good to me.

slarticodefast commented on Resources/Textures/Clothing/Head/Hardsuits/paramedhelm.rsi/meta.json:
```diff
@@ -1,7 +1,7 @@
   "version": 1,
   "license": "CC-BY-SA-3.0",
-  "copyright": "Taken from paradise station at commit https://github.com/ParadiseSS13/Paradise/commit/e5e584804b4b0b373a6a69d23afb73fd3c094365, redrawn by Ubaser. vox state by Flareguy",
+  "copyright": "Taken from paradise station at commit https://github.com/ParadiseSS13/Paradise/commit/e5e584804b4b0b373a6a69d23afb73fd3c094365, redrawn by Ubaser. vox state by Flareguy. flash states by Velken",
```
```suggestion
  "copyright": "Taken from paradise station at commit https://github.com/ParadiseSS13/Paradise/commit/e5e584804b4b0b373a6a69d23afb73fd3c094365, redrawn by Ubaser. vox state by Flareguy, icon-flash and light-overlay states by Velken",
```

slarticodefast commented on Resources/Prototypes/Entities/Clothing/OuterClothing/softsuits.yml:
```diff
@@ -89,41 +89,3 @@
   - type: HeldSpeedModifier
-
-#Paramedic Voidsuit
-#Despite having resistances and looking like a hardsuit, this parents off the EVA suit so it goes here.
```
It would probably make more sense to keep it in this file. Like the original comment says, it is not about being a hardsuit or softsuit, but about it inheriting from the EVA suit, so you can compare the yaml easier when they are in the same file.

slarticodefast commented on Resources/Audio/Effects/Vehicle/attributions.yml:
```diff
@@ -19,3 +19,8 @@
   license: "CC0-1.0"
   copyright: "13gkopeckak on freesound.org"
+
+- files: ["ambulancesiren.ogg"]
```
I'm not really a fan of the new siren sound either. It sounds like an ambulance is driving down the road outside my house, not like part of the game. Are there maybe any other sound effects on freesound?

Velken replied:
i've spent several hours already looking, but i can try looking more into another one

Velken commented:
changed siren sound again, made the suggested changes

Velken commented:
Final (pinky promise) change to the sound, recorded with game audio and PC audio at 100%
If you don't like it, please, by all means, go and make a PR to offer a better sound.
Unless you think is too quiet/loud, that is just a simple edit to the volume paramater in the softsuit yaml

https://github.com/user-attachments/assets/9be84442-e4ac-404b-8490-7d86b2452e76

slarticodefast commented on Resources/Prototypes/Entities/Clothing/Head/hardsuit-helmets.yml:
```diff
@@ -36,6 +36,8 @@
     - state: icon
     - state: icon-unshaded
       shader: unshaded
+    - state: icon-flash
+      visible: false
```
is this one even used?
the toggle seems to use the "light" map below

slarticodefast commented on Resources/Prototypes/Entities/Clothing/Head/hardsuit-helmets.yml:
```diff
@@ -95,6 +101,8 @@
     - state: icon
     - state: icon-unshaded
       shader: unshaded
+    - state: icon-flash
+      visible: false
```
same here

slarticodefast commented on Resources/Prototypes/Entities/Clothing/Head/hardsuit-helmets.yml:
```diff
@@ -148,6 +160,8 @@
     - state: icon
     - state: icon-unshaded
       shader: unshaded
+    - state: icon-flash
+      visible: false
```
same

slarticodefast commented on Resources/Prototypes/Entities/Clothing/Head/hardsuit-helmets.yml:
```diff
@@ -689,6 +755,8 @@
     sprite: Clothing/Head/Hardsuits/cburn.rsi
     layers:
     - state: icon
+    - state: icon-flash
+      visible: false
```
same

walksanatora commented:
I like that low-tone sound. it wont destroy my eardrums when I inevibitaly leave it on for hours on end.

slarticodefast commented on Resources/Prototypes/Entities/Clothing/Head/eva-helmets.yml:
```diff
@@ -50,17 +50,39 @@
   - type: Sprite
-    sprite: Clothing/Head/Helmets/paramedhelm.rsi
+    sprite: Clothing/Head/Hardsuits/paramedhelm.rsi
+  - type: HandheldLight
+    addPrefix: false
```
```suggestion
```
looks like this is not needed

slarticodefast reviewed (approved):
Looks good to me, thank you for your contribution!

slarticodefast commented on Resources/Prototypes/Entities/Clothing/Head/eva-helmets.yml:
```diff
@@ -50,17 +50,37 @@
   - type: Sprite
-    sprite: Clothing/Head/Helmets/paramedhelm.rsi
+    sprite: Clothing/Head/Hardsuits/paramedhelm.rsi
+  - type: ToggleableLightVisuals
```
```suggestion
  - type: HandheldLight
    addPrefix: false
  - type: ToggleableLightVisuals
```

beck-thompson reviewed (approved):
Looks good! The noise is fun :partying_face:

Welldzhik commented:
I am amused by the need for a sound signal for a space suit designed to work in an airless environment. Fortunately, I've already come to terms with playing the harmonica)
Unless I would like independent switches for sound and light.
